src/main/java/org/terasology/spawning/SpawnerSystem.java

```diff
@@ -367,21 +375,85 @@ public void update(float delta) {
 
                 // Temp hack - make portal spawned fancy mobs bounce around like idiots too just so they do something
                 // TODO: Change around so the Spawnable defines this
-                SimpleAIComponent simpleAIComponent = new SimpleAIComponent();
+                /*SimpleAIComponent simpleAIComponent = new SimpleAIComponent();
                 newSpawnableRef.addComponent((simpleAIComponent));
                 //newSpawnableRef.saveComponent(simpleAIComponent);
 
-                SpawnableComponent newSpawnable = newSpawnableRef.getComponent(SpawnableComponent.class);
-                newSpawnable.parent = entity;
+
+*/
+                List<EntityRef> spawnerEntities = Lists.newArrayList();
+
+                for (EntityRef spawner : entityManager.getEntitiesWith(SpawnerComponent.class, LocationComponent.class)) {
+
+                        spawnerEntities.add(spawner);
+                }
+
+                logger.info("Count of valid (also have a Location) Spawner entities: {}", spawnerEntities.size());
+                for (EntityRef entity : spawnerEntities) {
+                    SpawnerComponent spawnerComp = entity.getComponent(SpawnerComponent.class);
+
+                    Vector3f originPos = entity.getComponent(LocationComponent.class).getWorldPosition();
+                    Vector3f spawnPos = originPos;
+
+                    Prefab chosenPrefab = spawnerComp.requiredPrefab;
```

Comment:
I implemented logic to get the spawn prefab from the `PortalSystem` class by adding a variable to the `SpawnerComponent`, but this is still buggy and the `chosenPrefab` variable is always set to the fallback. This type of implementation would render `spawnTypes`, `spawnerType` and other such variables needless.